lib/list.js

```diff
@@ -64,7 +64,9 @@ function List(items, itemType, parent) {
   }
 
   List.prototype.toItem = function(item) {
-    return isClass(this.itemType) ? new this.itemType(item) : this.itemType(item);
+    return (isClass(this.itemType) || (this.itemType.toString() == Date.toString())) ?
```

Comment:
Let's extract this condition into a named variable to keep the code simpler to read.

```js
const isClassLike = isClass(this.itemType) || this.itemType.toString() == Date.toString();
return isClassLike ? new this.itemType(item) : this.itemType(item);
```

Also, do we need to convert `Date` constructor to string? Isn't it enough to compare the functions?

```js
const isClassLike = isClass(this.itemType) || this.itemType == Date;
```